lib/src/source/git.dart

```diff
@@ -426,6 +433,22 @@ class BoundGitSource extends CachedSource {
     _updatedRepos.add(path);
   }
 
+  /// Check if [directory] is a valid git repo. If invalid and
+  /// [delIfInvalid] is set to be true, it will be deleted.
+  ///
+  bool _checkIfValidGitRepo(Directory directory, {bool delIfInvalid = false}) {
```

Comment:
The return value is never used... And you only ever call it with `delIfInvalid: true`

How about renaming this to `_cleanInvalidGitRepository`?

Reply:
Since all invocations pass a `String` wrapped as `Directory`, how about just passing a `String`?

And maybe using `dirExists` instead of `Directory.existsSync()`, I assume the `lib/src/io.dart` has `dirExists` defined because it's easier to use :)